### ISSUE TITLE
Game should retain most recent score

### ISSUE THREAD

**Issue by phet-steele** [phetsims/graphing-lines#61]
The level selection screen needs to be changed to display stars for the most recent score achieved in a level. Currently the sim will always display the high score. Not a high priority issue, no need for a maintenance release. It's okay to wait until next re-deploy. See https://github.com/phetsims/balancing-act/issues/63 for the original issue.

**Comment by pixelzoom** [comment]
Deferred until phetsims/vegas#51 is resolved.

**Comment by pixelzoom** [comment]
phetsims/vegas#51 was resolved, so nothing blocking this.

**Comment by pixelzoom** [comment]
Before working on this... Review the resolution of https://github.com/phetsims/balancing-act/issues/63, which I don't understand.

**Comment by pixelzoom** [comment]
This applies to the game in both Graphing Lines and Graphing Slope-Intercept. @amanda-phet should I address this as part of graphing-slope-intercept development?

**Comment by amanda-phet** [comment]
@phet-steele Why does it need to show the most recent score? Some games show most recent, some show best (I learned this when working on different sim primers with different rules).

**Comment by phet-steele** [comment]
> Some games show most recent, some show best

@amanda-phet our sims are inconsistent only because some were published before a consensus was made. See the consensus at: https://github.com/phetsims/vegas/issues/51#issuecomment-207081290. The decision is that all games will show recent score.

With regards to this issue, it was made because of https://github.com/phetsims/balancing-act/issues/63#issuecomment-204021601

**Comment by pixelzoom** [comment]
I'm confused about what does or doesn't need to be done here.  Just wanted to see whether I need to look into it as part of graphing-slope-intercept.  Sounds like the answer is "yes", so self assigning.

**Comment by pixelzoom** [comment]
@amanda-phet I will be sure to consult with you when I figure out what needs to be done here.

**Comment by pixelzoom** [comment]
In https://github.com/phetsims/balancing-act/issues/63#issuecomment-204021601, @ariel-phet  wrote:
> At status meeting, @kathy-phet said that retaining most recent score should become our standard.

... and that was already the behavior in Balancing Act, so that issue was simply closed. 

Imo, this is decision if very odd, and (perhaps) warrants further discussion.  In (most?) games, your high score is shown. And beating your high score serves as motivation for continuing to play the game.  Showing most-recent score means that you'll have to remember your high score.  And if you miss a challenge, you'll be more likely to press the "Start Over" button than to continue playing the current game - is that behavior PhET wants to encourage? 

@amanda-phet how do you want to proceed?

**Comment by amanda-phet** [comment]
I agree with you @pixelzoom and am not sure what to do. Perhaps @ariel-phet or @kathy-phet could elaborate on the reasoning behind this decision? I see in the issue that @ariel-phet mentioned it could be unclear to the user because the sim does not specifically say "best" score, but we can change that if showing the best score is desired. I don't see an argument for showing the most recent score. @kathy-phet can you please explain the decision?

**Comment by ariel-phet** [comment]
@pixelzoom @amanda-phet my recollection of the rationale was classroom use. If a teacher is walking around seeing how a student is doing or wants them to replay a level they want to see the most recent score...or if the sim was used in a previous class and not reset. That is my recollection.

**Comment by amanda-phet** [comment]
The classroom use scenario isn't that clear to me... I guess I don't understand why a teacher would care to see the most recent score over the best score. Perhaps @kathy-phet will remember since she made the final call.

**Comment by pixelzoom** [comment]
Also important in this decision is that Graphing Lines (and other sims) display not only best score, but best _time_.  Turn on the timer, and you add the ability to play "beat the clock".   This provides an additional motivation for extending game play.  If we standardize on showing most-recent score, then presumably we are also going to show most-recent time.  And that doesn't support "beat the clock" game play at all.  

Changing the semantics also affects vegas APIs (PhET's repo of game components).   And while the decision to change to "most-recent" semantics was made almost a year ago, I see that no one has updated the vegas APIs.  Examples:

• `LevelSelection` button has options `bestTimeProperty`, `bestTimeVisibleProperty`, `bestTimeFill`, `bestTimeFont`, `bestTimeYSpacing`.
• `LevelCompletedNode` refers to `yourNewBestString`, `isNewBestTime`, `bestTimeAtThisLevel`, etc.
• The string file for vegas contains `"Your New Best!"` and `"Your Best: {0}"` (e.g. "Your Best (1:45)").   Do these need to be changed?  And re-translated?

Many sims keep track of "best", including area-builder, arithmetic, balancing-act, balancing-chemical-equations, build-an-atom, fraction-matcher, graphing-lines, and reactants-products-and-leftovers.  What affect will changing semantics have on these sims?

**Comment by pixelzoom** [comment]
As noted in https://github.com/phetsims/balancing-act/issues/63#issuecomment-288160897, at least one simulation has change the semantics without changing the names used in the code. Balancing Act still refers to "best score" throughout the code, but now stores the most-recent score.  I'm noting this here to make the point that changing semantics requires changing code.

**Comment by pixelzoom** [comment]
In summary, using "most-recent" semantics has 3 big cons: 
(1) Negative impact on extending game play, especially when the timer option is used.
(2) Requires significant changes to common-code APIs (vegas)
(3) Requires significant changes to simulations.

I have yet to hear what the pros are, and whether they outweigh the cons. @kathy-phet still awaiting your input on this.

@amanda-phet It would be nice if this were resolved before I return from vacation, since this is the final issue  to resolve before dev testing of graphing-slope-intercept begins.

**Comment by amyh-phet** [comment]
Not sure if it makes a difference, but if we decide to change this implementation, we will need to redo any teacher primers that reference game mechanics that have been changed.

**Comment by ariel-phet** [comment]
Just my 2 cents - I am not sure what my views on this issue were in the past, but currently I am fairly agnostic. We seem to get no complaints from users about one direction or the other. If it is easier (in terms of no code changes) perhaps we should just stick with the current approach, but have a definite choice of consistency going forward.

**Comment by kathy-phet** [comment]
I've reviewed the issue. Most games that I am aware of that show high score also show "your score" with respect to the high score, with high score being labeled. It also seems like when there is high score shown, there is often a social component where you are playing against a group of other players (classmates or such, and the person with the high score gets to have their name there). So our games are a bit different than that.

I'm not opposed to tracking high score, but it seems like we would want it to more clearly be labeled HS and to also track last score and even number of games played. This seems like quite a bit of design work. 

My inclination would be to leave the behavior as is -- since as Ariel said we don't currently get complaints -- in order to move forward with the current work. And then decide if we should enhance the game level screen and the priority of that. 

I recall being told that in interviews students like to get all the points, and what they often do is just hit restart if they get one wrong. I don't really see this behavior changing regardless of the way we record the scores. 

A few sims pose challenging issues. In Fraction Matcher and Arithmetic, you can go in and out of the level without restarting .... which I think is an important feature for some of these games ... and you can review your work after you complete the level which is also nice for Arithmetic. These variations would all need to be re-examined in order to come up with unifying behavior that supports the needs across these variations. 

While it would be nice to address this issue, it doesn't seem like a high priority given everything else on our list, so I recommend deferring this design task and just keep graphing lines behavior the same across its subset of sims (that is don't change it from its current graphing line behavior so we don't incur additional work at this time.)

**Comment by pixelzoom** [comment]
I thought it would be a good time to address it since (a) I just made other major changes to the Game, (b) the Game will need to be throughly retested in a related sims, and (c) graphing-slope-intercept is way ahead of schedule.

But it sounds like this is not the time that people want to address this.  So I'll unassign and mark as deferred.

**Comment by pixelzoom** [comment]
Btw... Best time is very clearly labeled as such on the screen that is shown when a game is completed.  For example in RPAL:

When you get a best time:

![screenshot_317](https://cloud.githubusercontent.com/assets/3046552/24671972/7ba3a18a-1930-11e7-876d-46da2bcfb9f7.png)

And when you beat your best time:

![screenshot_318](https://cloud.githubusercontent.com/assets/3046552/24671973/7d01412c-1930-11e7-80fb-738453c53700.png)

**Comment by pixelzoom** [comment]
Btw... Best time is very clearly labeled as such on the screen that is shown when a game is completed.  

For example in RPAL, when you beat your best time:

![screenshot_317](https://cloud.githubusercontent.com/assets/3046552/24671972/7ba3a18a-1930-11e7-876d-46da2bcfb9f7.png)

And when you don't beat your best time:

![screenshot_315](https://cloud.githubusercontent.com/assets/3046552/24672059/c73e9064-1930-11e7-8115-b47629030b87.png)

And the best time is shown on the level-selection screen:

![screenshot_319](https://cloud.githubusercontent.com/assets/3046552/24672100/e71ee8d4-1930-11e7-9d8b-20ef7f42aa8d.png)

**Comment by amanda-phet** [comment]
@phet-steele and I discussed this a bit yesterday and in the level completion panel (screenshotted above for RPAL) we could easily display "(Your Best: 10 out of 10)" below "Score: 10 out of 10". Then there is a nice parallel between the score and the time, and when you return to the level selection screen it's pretty clear what is being displayed. 

I think deferring the decision makes sense for the moment since a larger discussion about different types of game-play might need to happen first.

**Comment by kathy-phet** [comment]
> I thought it would be a good time to address this since (a) I just made other major changes to the Game, (b) the Game will need to be throughly retested in a related sims, and (c) graphing-slope-intercept is way ahead of schedule.

Thanks for watching out for this opportunity, Chris! But yes, I think I'd still like to defer given all the people that would need to be pulled into figure out a path forward for improving the design.

Regarding Graphing-Slope-Intercept being way ahead of schedule -- that is great news. The sooner it is out the better! :) Thanks.

**Comment by pixelzoom** [comment]
@amanda-phet Since we're going to publish a new minor release (1.3.0) of this sim, we're supposed to review all open issues.  Please evaluate whether you want to open this can of worms, or defer until a future release.

**Comment by amanda-phet** [comment]
What is the expected timeline for re-releasing GL? This is a good reminder that we need to have a design meeting about game dialogs.

**Comment by pixelzoom** [comment]
I don't think we have a timeline.  There are a bunch of changes that are "fixed pending deploy", and @ariel-phet said it would be good to publish a new minor release sometime.   @ariel-phet?

**Comment by pixelzoom** [comment]
Discussed with @amanda-phet, we're going to defer this for 1.3.0 release.

**Comment by pixelzoom** [comment]
@ariel-phet I'm guessing that the goal is to release 1.3.0 of this sim shortly after Graphing Quadratics 1.0, to take advantage of the improvements we've made to make this sims more similar.  Is that correct?

**Comment by kathy-phet** [comment]
I'd favor deferring this particular issue, as I think it will take some time and we have some other higher priority items.

**Comment by pixelzoom** [comment]
Yep, we're deferring, see https://github.com/phetsims/graphing-lines/issues/61#issuecomment-436057456.  @amanda-phet is going to put the general issue somewhere on her TODO list.

**Comment by pixelzoom** [comment]
Deferred until phetsims/vegas#51 is resolved.

**Comment by pixelzoom** [comment]
In 12/15/23 design meeting, we decided to stick with showing best score.  See notes in https://github.com/phetsims/vegas/issues/51#issuecomment-1858343945.

So closing this issues as "won't fix".
